lib/selinuxtest.pm

```diff
@@ -93,4 +96,57 @@ sub check_category {
     }
 }
 
+sub set_sestatus {
+    my ($self, $mode, $type) = @_;
+    my $selinux_config_file = "/etc/selinux/config";
+    select_console "root-console";
+
+    # SELinux by default
+    validate_script_output("sestatus", sub { m/SELinux status: .*disabled/ });
+
+    # workaround for "selinux-auto-relabel" in case: auto relabel then trigger reboot
+    my $results = script_run("zypper --non-interactive se selinux-autorelabel");
+    if (!$results) {
+        assert_script_run("sed -ie \'s/GRUB_TIMEOUT.*/GRUB_TIMEOUT=8/\' /etc/default/grub");
+    }
+
+    # enable SELinux in grub
+    if ($mode eq "enforcing") {
+        add_grub_cmdline_settings('security=selinux selinux=1 enforcing=1', update_grub => 1);
+    }
+    elsif ($mode eq "permissive") {
+        add_grub_cmdline_settings('security=selinux selinux=1 enforcing=0', update_grub => 1);
+    }
```

Comment:
This means that selinux is not enabled unless the mode is either "enforcing" or "permissive" so the code is doing something different than the comment says. If you want to be strict that either value is set I suggest

```suggestion
    die "Need mode 'enforcing' or 'permissive'" unless $mode =~ /enforcing|permissive/;
    add_grub_cmdline_settings('security=selinux selinux=1 enforcing=' . $mode eq 'enforcing' ? 1 : 0, update_grub => 1);
```

Reply:
Done. Thanks.